pkg/build/builder/common.go

```diff
@@ -180,3 +204,23 @@ func execPostCommitHook(client DockerClient, postCommitSpec api.BuildPostCommitS
 		Stderr:       true,
 	})
 }
+
+func updateStatusRetrier(build *api.Build, client client.BuildInterface) error {
+	return kclient.RetryOnConflict(kclient.DefaultBackoff, func() error {
+		// before updating, make sure we are using the latest version of the build
+		latestBuild, err := client.Get(build.Name)
```

Comment:
@liggitt  @deads2k 
can you tell me how i can get the latest build object from the API from the builder itself?
i've tried something below with adding a bootstrap policy for serviceaccount:builder but it doesn't seem to work. i need to get the latest version of the build object.

the above code fails with 
>User "system:serviceaccount:openshift:builder" cannot get builds in project "openshift"
also the above code executes here: https://github.com/openshift/origin/pull/10817/files#diff-c63f8d7a6dbdd8d953d6bd2afed3330bR72

i tried adding a serviceaccount:builder like this:
https://github.com/openshift/origin/pull/10817/files#diff-1d38b34f638332aebf97978ff71d9664R453
https://github.com/openshift/origin/pull/10817/files#diff-1d38b34f638332aebf97978ff71d9664R129

i also tried adding `get` a rule here: 
https://github.com/openshift/origin/blob/master/pkg/cmd/server/bootstrappolicy/policy.go#L447
the problem is that that's the ImageBuilderRoleName      = "system:image-builder"
not system:builder or something similar.
i'm quite convinced this is not the way to do it, but i don't know what else to try.